src/Controls/src/Core/Handlers/Items/iOS/TemplatedCell.cs

```diff
@@ -89,39 +94,35 @@ public override UICollectionViewLayoutAttributes PreferredLayoutAttributesFittin
 		{
 			var preferredAttributes = base.PreferredLayoutAttributesFittingAttributes(layoutAttributes);
 
-			var preferredSize = preferredAttributes.Frame.Size;
-
-			if (preferredSize.IsCloseTo(_size)
-				&& AttributesConsistentWithConstrainedDimension(preferredAttributes))
+			if (_measureInvalidated || !AttributesConsistentWithConstrainedDimension(preferredAttributes))
 			{
-				return preferredAttributes;
+				UpdateCellSize();
 			}
 
-			var size = UpdateCellSize();
-
 			// Adjust the preferred attributes to include space for the Forms element
-			preferredAttributes.Frame = new CGRect(preferredAttributes.Frame.Location, size);
+			preferredAttributes.Frame = new CGRect(preferredAttributes.Frame.Location, _size);
+
+			// We need to set the Frame on the virtual view.
+			// Subviews will eventually be arranged via LayoutSubviews.
+			var frame = new Rect(Point.Zero, _size);
+			var virtualView = PlatformHandler.VirtualView!;
```

Comment:
I'm not a huge fan of `!` operator, because it may be harder to find the NRE latter. The .NET SDK team uses the `Debug.Assert` for that. I think it make sense to use here. What do you think?

```suggestion
			var virtualView = PlatformHandler.VirtualView;
			Debug.Assert(virtualView is not null, "VirtualView shouldn't be null here");
```

Reply:
I think I'm gonna explicitly throw an exception as the virtual view must be there in order to perform measure/arrange.